scripts/auto-rebase/presubmit.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import glob
+import yaml
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+ASSETS_DIR = "assets/"
+STAGING_DIR = "_output/staging/"
+
+def build_assets_filelist_from_asset_dir(dir, prefix=""):
+    assets_files = []
+
+    dir_path = os.path.join(prefix, dir['dir'])
+    if 'files' in dir:
+        for file in dir['files']:
+            assets_files.append(os.path.join(dir_path, file['file']))
+
+    if 'dirs' in dir:
+        for subdir in dir['dirs']:
+            assets_files.extend(build_assets_filelist_from_asset_dir(subdir, dir_path))
+
+    return assets_files
+
+
+def build_assets_filelist_from_recipe(recipe):
+    assets_files = []
+
+    for asset in recipe['assets']:
+        if "dir" in asset:
+            assets_files.extend(build_assets_filelist_from_asset_dir(asset))
+
+        if 'file' in asset:
+            assets_files.append(asset['file'])
+
+    return assets_files
+
+
+def main():
+    if not os.path.isdir(ASSETS_DIR):
+        logging.error(f"Expected to run in root directory of microshift repository but was in {os.getcwd()}")
+        sys.exit(1)
+
+    if not os.path.isdir(STAGING_DIR):
+        logging.error(f"{STAGING_DIR} does not exist")
+        sys.exit(1)
+
+    recipe_filepath = "./scripts/auto-rebase/assets.yaml"
+    recipe = yaml.load(open(recipe_filepath).read(), Loader=Loader)
+    assets_filelist = build_assets_filelist_from_recipe(recipe)
+
+    realfiles =  [ f.removeprefix('assets/') for f in glob.glob('assets/**/*.*', recursive=True) ]
+    missing = []
+    for f in realfiles:
+        if f not in assets_filelist:
```

Comment:
Should we check for the reverse condition (where the file is in the asset list but not the repo), too?

Reply:
Great idea, done!